docs/reference/connectors-kibana/notion-action-type.md

```diff
@@ -0,0 +1,70 @@
+---
+navigation_title: "Notion"
+mapped_pages:
+  - https://www.elastic.co/guide/en/kibana/current/notion-action-type.html
+applies_to:
+  stack: preview
+  serverless: preview
+---
+
+# Notion connector [notion-action-type]
+
+The Notion connector communicates with the Notion API to explore content and databases in your Notion workspace.
+
+## Create connectors in {{kib}} [define-notion-ui]
+
+You can create connectors in **{{stack-manage-app}} > {{connectors-ui}}**. For example:
+
+### Connector configuration [notion-connector-configuration]
+
+Notion connectors have the following configuration properties:
+
+API Token
+:   The Notion API token (bearer token) for authentication.
+
+## Test connectors [notion-action-configuration]
+
+You can test connectors as you're creating or editing the connector in {{kib}}.
+
+The Notion connector has the following actions:
+
+Search Page or Data Source by Title
+:   Search for pages or databases by title.
+    - **Query** (required): The search query string.
+    - **Query Object Type** (required): Type of object to search for (`page` or `data_source`).
+    - **Start Cursor** (optional): Cursor for pagination.
+    - **Page Size** (optional): Number of results per page.
+
+Get Page
+:   Retrieve a page by its ID.
+    - **Page ID** (required): The unique identifier of the page.
+
+Get Data Source
+:   Retrieve a database by its ID.
+    - **Data Source ID** (required): The unique identifier of the database.
+
+Query Data Source
+:   Query a database with optional filters.
+    - **Data Source ID** (required): The unique identifier of the database to query.
+    - **Filter** (optional): JSON string representing the filter object.
+    - **Start Cursor** (optional): Cursor for pagination.
+    - **Page Size** (optional): Number of results per page.
+
+## Connector networking configuration [notion-connector-networking-configuration]
+
+Use the [Action configuration settings](/reference/configuration-reference/alerting-settings.md#action-settings) to customize connector networking configurations, such as proxies, certificates, or TLS settings. You can set configurations that apply to all your connectors or use `xpack.actions.customHostSettings` to set per-host configurations.
```

Comment:
I had no idea about this, so it's not tested. We can leave it out, or if it's something that's meant to come for free for everything that implements `ConnectorSpec` interface, then leave it in and come back to it if a user raises any issues.